python/python/glide/async_commands/core.py

```diff
@@ -392,6 +392,40 @@ async def _cluster_scan(
         type: Optional[ObjectType] = ...,
     ) -> TResult: ...
 
+    async def _reset_connection_password(
+        self, password: Optional[str], re_auth: bool
+    ) -> TResult: ...
+
+    async def reset_connection_password(
+        self, password: Optional[str], re_auth: bool
+    ) -> TOK:
+        """
+        Reset the current connection password with a new password.
+
+        **Note:** This method updates the client's internal password configuration and does
+        not perform password rotation on the server side.
+
+        This method is useful in scenarios where the server password has changed or when
+        utilizing short-lived passwords for enhanced security. It allows the client to
+        update its password to reconnect upon disconnection without the need to recreate
+        the client instance. This ensures that the internal reconnection mechanism can
+        handle reconnection seamlessly, preventing the loss of in-flight commands.
+
+        Args:
+            password (Optional[str]): The new password to use for the connection, if None the password will be removed.
+            re_auth (bool):
```

Comment:
I propose to have an enum for that, because the options aren't contraditory as true/false values.

Reply:
Explain?